Detectors/Calibration/src/MeanVertexCalibrator.cxx

```diff
@@ -35,137 +36,280 @@ void MeanVertexCalibrator::initOutput()
-      }
+      LOG(info) << "SigmaX = " << fitResSlicesX[iFit][2] << " error = " << covMatrixX[iFit](2, 2);
+      LOG(info) << "SigmaY = " << fitResSlicesY[iFit][2] << " error = " << covMatrixY[iFit](2, 2);
     }
-    std::vector<float> fitValues;
-    float* array = &mSMAdata.histoX[0];
-    double fitres = fitGaus(mSMAdata.nbinsX, array, -(mSMAdata.rangeX), mSMAdata.rangeX, fitValues);
-    if (fitres >= 0) {
-      LOG(info) << "X: Fit result (of merged Slots) => " << fitres << ". Mean = " << fitValues[1] << " Sigma = " << fitValues[2];
-    } else {
-      LOG(error) << "X: Fit failed with result = " << fitres;
+    if (covMatrixX[iFit](2, 2) != 0) {
+      double weightSigma = 1. / covMatrixX[iFit](2, 2); // covMatrix is already an error squared
+      sumX += (fitResSlicesX[iFit][2] * weightSigma);
+      weightSumX += weightSigma;
+    }
+    if (covMatrixY[iFit](2, 2) != 0) {
+      double weightSigma = 1. / covMatrixY[iFit](2, 2); // covMatrix is already an error squared
+      sumY += (fitResSlicesY[iFit][2] * weightSigma);
+      weightSumY += weightSigma;
     }
-    mSMAMVobj.setX(fitValues[1]);
-    mSMAMVobj.setSigmaX(fitValues[2]);
+  }
+  if (mVerbose) {
+    LOG(info) << "sumX = " << sumX;
+    LOG(info) << "weightSumX = " << weightSumX;
+    LOG(info) << "sumY = " << sumY;
+    LOG(info) << "weightSumY = " << weightSumY;
+  }
+
+  double sigmaX = 0;
+  if (weightSumX != 0) {
+    sigmaX = sumX / weightSumX;
+  }
+  double sigmaY = 0;
+  if (weightSumY != 0) {
+    sigmaY = sumY / weightSumY;
+  }
+  if (mVerbose) {
+    LOG(info) << "SigmaX for MeanVertex = " << sigmaX;
+    LOG(info) << "SigmaY for MeanVertex = " << sigmaY;
+  }
+  mvo.setSigmaX(sigmaX);
+  mvo.setSigmaY(sigmaY);
 
-    // y coordinate
+  // now we get the slope for the x-coordinate dependence on z
+  TLinearFitter lf(1, "pol1");
+  lf.StoreData(kFALSE);
+  for (int i = 0; i < fitResSlicesX.size(); ++i) {
     if (mVerbose) {
-      LOG(info) << "**** Printing content of SMA MVData object for y coordinate";
-      for (int i = 0; i < mSMAdata.nbinsY; i++) {
-        LOG(info) << "i = " << i << ", content of histogram = " << mSMAdata.histoY[i];
-      }
+      LOG(info) << "Adding point " << i << ": zvtx = " << meanZvect[i] << " xvtx = " << fitResSlicesX[i][2];
     }
-    array = &mSMAdata.histoY[0];
-    fitres = fitGaus(mSMAdata.nbinsY, array, -(mSMAdata.rangeY), mSMAdata.rangeY, fitValues);
-    if (fitres >= 0) {
-      LOG(info) << "Y: Fit result (of merged Slots) => " << fitres << ". Mean = " << fitValues[1] << " Sigma = " << fitValues[2];
-    } else {
-      LOG(error) << "Y: Fit failed with result = " << fitres;
-    }
-    mSMAMVobj.setY(fitValues[1]);
-    mSMAMVobj.setSigmaY(fitValues[2]);
+    lf.AddPoint(&meanZvect[i], fitResSlicesX[i][1]);
+  }
+  lf.Eval();
+  double slopeX = lf.GetParameter(0);
+  mvo.setSlopeX(slopeX);
+  mvo.setX(mvo.getZ() * slopeX + lf.GetParameter(1));
```

Comment:
isn't it opposite: `offset = lf.GetParameter(0)` and `slope = lf.GetParameter(1)` ?

Reply:
Good catch, thanks!